hyper/contrib.py

```diff
@@ -43,28 +48,40 @@ def get_connection(self, host, port, scheme, cert=None):
         if cert is not None:
             ssl_context = init_context(cert=cert)
 
+        connection_key = (host, port, scheme, cert, proxy)
```

Comment:
Yeah, but frozenset of tuple isn't. You could write a transform that does `proxy_headers_key = frozenset((k, v) for k, v in proxy_headers.items())`.